_posts/2021-01-22-x-ways-to-write-meaningful-commit-messages.md

```diff
@@ -0,0 +1,226 @@
+- `feat`: introduces a new feature to the codebase (correlates with MINOR in semantic versioning)
+- `refactor!`: introduces a breaking API change by refactoring because of the **!** symbol (correlating with MAJOR in semantic versioning)
+  
+The symbol **"!"** can be used with any type. It signifies a breaking change that correlates with MAJOR in semantic versioning.
+
+Using `BREAKING CHANGE` in the footer introduces a breaking API change as well (correlating with MAJOR in semantic versioning).
+
+The [Angular commit message format](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit) is another conventional format. It suggests that a commit message should
+consist of a header, a body, and a footer with a blank line between each section because tools like `rebase` in Git 
+get confused if we run them together without space.
+
+```
+[type] [optional scope]: [short summary]
+
+[body] - at least 20 characters up to 72, optional only for docs
+
+[optional footer]
+```
+
+The header consists of a type and a summary part. Some add an optional "scope" in between.
+
+### Type
+The type of commit message says that the change was made for a particular problem. For example, if we've fixed a bug or added a feature, or maybe changed something related to the docs, the type would be "fix", "feat", or "docs".
+ 
+This format allows multiple types other than "fix:" and "feat:" mentioned in the previous part about conventional messages.
+Some other Angular's type suggestions are: "build:", "chore:", "ci:", "docs:", "style:", "refactor:", "perf:", "test:", and others.
+
+### Scope
+The scope is the package or module that is affected by the change. As mentioned before, it's optional.
+
+### Summary
+As Angular [suggests](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit): "It should be present tense. Not capitalized. No period in the end.", and **imperative like the type**.
+
+As Chris Beams mentions [in his article about commit messages](https://chris.beams.io/posts/git-commit/), the summary should always be able to complete the following sentence: 
+
+**If applied, this commit will...** add authorization for document access
+
+Let's look at some summary examples:
+```
+Right: fix: add authorization for document access
+Wrong: fix: Add authorization for document access (capitalized)
+Wrong: fix: added authorization for document access (not present tense)
+Wrong: fix: add authorization for document access. (period in the end)
+```
+
+In this example, "fix" is the type, and the sentence after that is the summary.
+
+### Body
+The format of the body should be just like the summary, but the content goal is different. It should explain the motivation for the change.
+
+In other words, **it should be an imperative sentence explaining why we're changing the code, compared to what it was before**.
+
+### Footer
+In the footer, we can mention the related task URL or the number of the issue that we worked on:
+
```

Comment:
I removed a section here (about a completely new format suggested by Chris Beams), because it didn't fit into the "Footer" section. Feel free to add it again, but it should go into the section "Conventional Commits Structure" where the other formats are explained, too.

Reply:
I didn't mean to put it there as part of the footer. It was supposed to be another conventional type which would come after Angular section that contains titles (type, scope, summary, body and footer). I think I didn't separate them correctly that it seemed to be under "Footer". But it's ok. I put the link to Chris Beams' convention at the end of the article. So thanks for the edit.